structuralcodes/codes/mc2020/_concrete_material_properties.py

```diff
@@ -0,0 +1,2833 @@
+    )
+
+
+def E_ci(fcm: float, alpha_E: float = 1.0, Ec0: float = 21500) -> float:
+    """Calculate the modulus of elasticity Eci for normal
+        weight concrete at 28 days.
+
+    fib Model Code 2020, eq. (14.6-20)
+
+    Args:
+        fcm (float): Mean compressive strength fck in MPa.
+        alpha_E (float): Coefficient depending on
+            the aggregate type (default is 1.0 for quartzite).
+        Ec0 (float): base elastic modulus value in MPa.
+
+    Returns:
+        float: Modulus of elasticity E_ci in MPa.
+
+    Raises:
+        ValueError: If fck or alpha_E is invalid.
+    """
+    if fcm <= 0:
+        raise ValueError(
+            'Characteristic compressive strength fck must be positive.'
+        )
+    if not (0.7 <= alpha_E <= 1.2):
+        raise ValueError('Coefficient alpha_E must be between 0.7 and 1.2.')
+
+    return alpha_E * Ec0 * (fcm / 10) ** (1 / 3)
+
+
+def El_ci(
+    fcm: float, density: float, alpha_E: float = 1.0, Ec0: float = 21500
+) -> float:
+    """Calculate the modulus of elasticity Elci for
+        lightweight aggregate concrete.
+
+    fib Model Code 2020, eq. (14.6-22)
+
+    Args:
+        fcm (float): Mean compressive strength fcm in MPa.
+        density (float): Oven-dry density of the lightweight
+            aggregate concrete in kg/m3.
+        alpha_E (float): Coefficient depending on the aggregate
+            type (default is 1.0 for quartzite).
+        Ec0 (float): base elastic modulus value in MPa.
+
+    Returns:
+        float: Modulus of elasticity El_ci in MPa.
+
+    Raises:
+        ValueError: If fcm, density, or alpha_E is invalid.
+    """
+    if fcm <= 0:
+        raise ValueError('Mean compressive strength fcm must be positive.')
+    if density <= 0:
+        raise ValueError('Density must be positive.')
+    if not (0.7 <= alpha_E <= 1.2):
+        raise ValueError('Coefficient alpha_E must be between 0.7 and 1.2.')
+
+    eta_E = (density / 2200) ** 2
+
+    Eci = alpha_E * Ec0 * (fcm / 10) ** (1 / 3)
+    return eta_E * Eci
+
+
+def E_ci_red_el(Eci: float, fcm: float) -> float:
+    """Calculate the reduced modulus of elasticity Ec for concrete
+        for only static analysis.
+
+    fib Model Code 2020, eq. (14.6-23) (14.6-24)
+
+    Args:
+        Eci (float): Modulus of elasticity Eci in MPa.
+        fcm (float): Mean compressive strength fcm in MPa.
+
+    Returns:
+        float: Reduced modulus of elasticity Ec in MPa.
+
+    Raises:
+        ValueError: If Eci or fcm is negative or zero.
+    """
+    if Eci < 0:
+        raise ValueError('Modulus of elasticity Eci must be positive.')
+
+    alpha_i = min(0.8 + 0.2 * fcm / 88, 1)
+    return alpha_i * Eci
+
+
+def nu_c() -> float:
+    """Get the Poisson's ratio νc for concrete for design purposes.
+
+    fib Model Code 2020 (14.6.1.4.3)
+
+    Returns:
+        float: Poisson's ratio νc for concrete,
+            typically 0.20 for design purposes.
+    """
+    return 0.20
+
+
+def sigma_c(
```

Comment:
This function should be considered removed in favour of one of our builtin constitutive laws. @talledodiego can you advice on this?